test/Npgsql.Tests/Support/PgPostmasterMock.cs

```diff
@@ -55,20 +64,17 @@ internal PgPostmasterMock(string? connectionString = null)
 
         void AcceptClients()
         {
+            _acceptingClients = true;
             _acceptClientsTask = DoAcceptClients();
 
             async Task DoAcceptClients()
             {
                 while (true)
                 {
                     var serverOrCancellationRequest = await Accept();
+                    await _pendingRequestsWriter.WriteAsync(serverOrCancellationRequest);
                     if (serverOrCancellationRequest.Server is { } server)
-                    {
-                        _pendingServers.Enqueue(server);
-                        await server.Startup();
-                    }
-                    else
-                        _pendingCancellationRequests.Enqueue(serverOrCancellationRequest.CancellationRequest!.Value);
+                        _ = server.Startup(); // Don't wait for this to complete
```

Comment:
Why?

Reply:
The postmaster should continue accepting new connections before the server starts up - note that Startup reads the startup packet from the client, so it might hang for a while... Do you see an issue here?

Reply:
Oh, so it's more of an optimization.

Reply:
It could be a real problem in some tests where you open multiple servers and don't want to wait for Startup to complete etc. (we don't have such a test currently, but we might in the future).